src/state/process.rs

```diff
@@ -407,7 +408,11 @@ where
         // If the proposal epoch ID is higher than the current epoch ID or proposal epoch ID is
         // equal to the current epoch ID and the proposal round is higher than the current round,
         // cache it until that epoch ID.
-        if epoch_id > self.epoch_id || (epoch_id == self.epoch_id && round > self.round) {
+        if (epoch_id > self.epoch_id && self.epoch_id + FUTURE_EPOCH_GAP > epoch_id)
```

Comment:
If epoch_id between (self.epoch_id, self.epoch_id + FUTURE_EPOCH_GAP)，but round is very large, it is also not allowed to insert into self.proposals. You should concern that. 
Besides, the proposals not satisfy your condition is not filtered.

Reply:
ok